src/ecs/resources/rendering.rs

```diff
@@ -0,0 +1,81 @@
+use crossbeam::sync::MsQueue;
+use futures::{Async, Future, Poll};
+use futures::sync::oneshot::{Receiver, channel};
+use gfx::traits::Pod;
+use renderer::{Error, Material, MaterialBuilder, Mesh, MeshBuilder};
+
+/// A factory future.
+pub struct FactoryFuture<A, E>(Receiver<Result<A, E>>);
+
+impl<A, E> Future for FactoryFuture<A, E> {
+    type Item = A;
+    type Error = E;
+
+    fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
+        match self.0.poll().expect("Sender destroyed") {
+            Async::Ready(x) => x.map(Async::Ready),
+            Async::NotReady => Ok(Async::NotReady),
+        }
+    }
+}
+
+/// The factory abstraction, which allows to access the real
+/// factory and returns futures.
+pub struct Factory {
+    jobs: MsQueue<Box<FnOnce(&mut ::renderer::Factory)>>,
```

Comment:
I guess not since it takes `self`, not `self: Box<Self>`; given that we cannot move unsized types, I should change that. Thanks!